packages/web/src/components/organisms/ExecutionsTable/ExecutionsFilters/ExecutionsFilters.tsx

```diff
@@ -0,0 +1,52 @@
+import {useEffect, useState} from 'react';
+import {useDebounce} from 'react-use';
+
+import {SearchOutlined} from '@ant-design/icons';
+
+import {useEntityDetailsField, useEntityDetailsPick} from '@store/entityDetails';
+
+import Colors from '@styles/Colors';
+
+import * as S from './ExecutionsFilters.styled';
+import ExecutionsStatusFilter from './ExecutionsStatusFilter';
+
+const ExecutionsFilters: React.FC = () => {
+  const [executionsFilters, setExecutionsFilters] = useEntityDetailsField('executionsFilters');
+  const {executionsLoading} = useEntityDetailsPick('executionsLoading');
+
+  const [searchInputValue, setSearchInputValue] = useState(executionsFilters.textSearch);
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchInputValue(e.target.value);
+  };
+
+  const [, cancel] = useDebounce(
+    () => {
+      setExecutionsFilters({...executionsFilters, textSearch: searchInputValue});
+    },
+    300,
+    [searchInputValue]
+  );
+
+  useEffect(() => {
```

Comment:
shouldn't we react to `searchInputValue` instead?

Reply:
We do that for the debounce, useEffect is purely just for the `cancel` method. Or am I missing your point? 🤔